packages/utils/src/logger.ts

```diff
@@ -97,4 +97,17 @@ function makeLogger(): Logger {
   return logger as Logger;
 }
 
-export const logger = makeLogger();
+/**
+ * This is a logger singleton which either logs things or no-ops if logging is not enabled.
+ * The logger is a singleton on the carrier, to ensure that a consistent logger is used throughout the SDK.
+ */
+export const logger: Logger = new Proxy(
```

Comment:
Actually, let me see if this is needed - I experimented a bit with this, maybe we can just do `const logger = getGlobalSingleton(...)` directly, skipping the proxy.